corporate/lib/stripe.py

```diff
@@ -1649,6 +1649,19 @@ def approve_sponsorship(self) -> None:
         # TBD
         pass
 
+    @override
+    def is_sponsored_or_pending(self, customer: Optional[Customer]) -> bool:
+        if (
+            customer is not None and customer.sponsorship_pending
+        ) or self.remote_realm.plan_type == self.remote_realm.PLAN_TYPE_COMMUNITY:
+            return True
+        return False
+
+    @override
+    def update_context_initial_upgrade(self, context: Dict[str, Any]) -> None:
+        # TBD
+        pass
```

Comment:
So I think the `get_initial_upgrade_context` override function shouldn't need to exist; the existing common one on the `BillingSession` abstract base class is intended to handle all the common code, with any special context specific to the view being prepared here, in `update_context_initial_upgrade`.

Small side note: The `billing_session_url` method above probably needs to be filled in with the right base URL in order for the redirects in `initial_upgrade` to work.

Reply:
Yeah, done.